crates/hyperdrive-math/src/long/open.rs

```diff
@@ -44,8 +46,15 @@ impl State {
     }
 
     /// Calculates the spot price after opening a Hyperdrive long.
-    pub fn calculate_spot_price_after_long(&self, base_amount: FixedPoint) -> FixedPoint {
-        let bond_amount = self.calculate_open_long(base_amount);
+    pub fn calculate_spot_price_after_long(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: Option<FixedPoint>,
+    ) -> FixedPoint {
+        let bond_amount = match bond_amount {
+            Some(bond_amount) => bond_amount,
+            None => self.calculate_open_long(base_amount),
+        };
         self.spot_price_after_long(base_amount, bond_amount)
```

Comment:
Now that this function can take a `bond_amount`, we can probably consolidate it with the `spot_price_after_long` function below to avoid confusion about which one to use.